fleet/fleet.go

```diff
@@ -131,18 +132,15 @@ type Fleet interface {
 func NewFleet(config Config) (Fleet, error) {
 	var trans http.RoundTripper
 
-	switch config.Endpoint.Scheme {
-	case "unix", "file":
-		if config.Endpoint.Host != "" {
-			// This commonly happens if the user misses the leading slash after the
-			// scheme. For example, "unix://var/run/fleet.sock" would be parsed as
-			// host "var".
-			return nil, maskAnyf(invalidEndpointError, "cannot connect to host %q with scheme %q", config.Endpoint.Host, config.Endpoint.Scheme)
+	tunneling := config.Tunnel.Tunnel != ""
+	// If a tunnel is provided we need to overwrite the http.Transport.Dial function
+	// to use the tunnel
+	if tunneling {
```

Comment:
no matter if the tunnel flag is set or not: if the endpoint is unix socket then we need to make some changes (https://github.com/giantswarm/inago/blob/ssh-tunnel/fleet/fleet.go#L161-L164) to the endpoint in the config.
Since this is always true for socket connections I feel we have two places in the code that do the same thing now